local-build.sh

```diff
@@ -73,23 +76,13 @@ function run_base() {
   if ! docker run -it --name yunikorn-site-local -d \
     -p 3000:3000 \
     -v "$PWD":/yunikorn-site \
+    --mount type=volume,dst=/yunikorn-site/node_modules \
```

Comment:
I think it would be better if we can explain the reason under run_base()'s comments. 
ex:
```
function run_base() {
  # run docker container
  # mount the repo root to the container working dir,
  # so that changes made in the repo can trigger the web-server auto-update
  # Alos prevent sharing node_modules/ between the host and the Docker environment to avoid potential conflict.
```

Reply:
Docker behavior is well documented already bud rather not clutter a simple function with a large comment here.

Reply:
> It isolated the host's and docker's node_modules/ . 
> 
> Could I confirm what is the previous weirdness in the old build? I wasn't aware of it before.
> 
> 
> 
> > Most of the weirdness in the old build was due to node_modules being (incorrectly) shared between the host and the Docker environment.

We had all sorts of strange brokenness around algolia search having to be reinstalled constantly. I suspect there was platform-specific code being added and this caused issues when node_modules was shared between different platforms.